### ISSUE TITLE
SD DMA Timeout: running tagdump against gps collector tag eventually msg timeout

### ISSUE THREAD

**Issue by cire831** [MamMark/mm#41]
looks like the msg time out comes from a SD_WRITE_DMA timeout which panics.

**Comment by cire831** [comment]
need instrumentation that captures dma state when the timeout occurs.

we looked at dma state after the panic (human time) and no dma is pending.

**Comment by cire831** [comment]
Implemented DMA failure instrumentation in 91f1029

We bumped the SD DMA Xfer timeout from (4 * 8) to (4 * 8 * 8) in 0859d76, and we haven't seen any failures since then.  Read the comments in tos/chips/sd/sd.h.  We are currently running the main DCO MCLK at 16MiHz and the SMCLK at 16MiHz/2 = 8MiHz.  We calculate the transfer time for 512 bytes to be 4mS.  We first bumped that up by a factor of 8 for 32 mis which should have been enough but we would intermittently get the DMA timeout failures.

Would be nice to better understand what is happening but for now the larger deadman seems to work fine.  Very weird.

Got a number of failures with the timeout set to (4 * 8).  The SDTimer is expiring in a SD DMA state.  The capture shows the DMA has completed at the time of the capture code executing (at the time the timer goes off).  This says that we are either losing the interrupt and post of the dma task or the DMA has completed after the Timing code has posted its task.  The dma task will then be queued after the Timing task.

We should probably modify the SDTimer code to check for a dma completion and/or dma task is pending.  But upping the timeout has the same effect.

since we bumped the timeout to  (4 * 8 * 8) we haven't seen any other failures from the SD DMA code timing out.
